cmds/rm/rm.go

```diff
@@ -11,81 +11,142 @@
 //     -i: interactive mode
 //     -v: verbose mode
 //     -R: remove file hierarchies
-//     -r: equivalent to -R
 package main
 
 import (
 	"bufio"
 	"flag"
 	"fmt"
+	"io/ioutil"
 	"os"
-	"path"
+	"path/filepath"
+	"syscall"
+	"strings"
+	"log"
 )
 
-var (
-	flags struct {
-		r bool
-		v bool
-		i bool
-	}
-	cmd = "rm [-Rrvi] file..."
-)
+// You can add more flags to this struct.
+type rmFlags struct {
+	recursive   bool
+	verbose     bool
+	interactive bool
+}
 
-func init() {
-	defUsage := flag.Usage
-	flag.Usage = func() {
-		os.Args[0] = cmd
-		defUsage()
+// Helper method for interactive flag
+func iFlag(input *bufio.Scanner, printString string) bool {
+	for{	
+		//replace Println with Printf
+		fmt.Printf("%s (y/n)\n", printString)
+		input.Scan()
+		if input.Err()!= nil{
+			log.Fatalf("Failed at error: %v", input.Err())		
+		}
+		if(strings.Compare(input.Text(), "y") == 0){	
+			return true
+		}
+		if(strings.Compare(input.Text(), "n") == 0){
```

Comment:
could also do this:
switch input.Text() {
case "yes":
return true
case "no":
return false
}